code/game/antagonist/antagonist.dm

```diff
@@ -116,46 +117,57 @@
 	return candidates
 
 /datum/antagonist/proc/attempt_random_spawn()
+	update_current_antag_max()
 	build_candidate_list(flags & (ANTAG_OVERRIDE_MOB|ANTAG_OVERRIDE_JOB))
 	attempt_spawn()
 	finalize_spawn()
 
-/datum/antagonist/proc/attempt_late_spawn(var/datum/mind/player)
+/datum/antagonist/proc/attempt_auto_spawn()
 	if(!can_late_spawn())
 		return 0
-	if(!istype(player))
-		var/list/players = get_candidates(is_latejoin_template())
-		if(players && players.len)
-			player = pick(players)
-	if(!istype(player))
-		message_admins("[uppertext(ticker.mode.name)]: Failed to find a candidate for [role_text].")
-		return 0
-	player.current << "<span class='danger'><i>You have been selected this round as an antagonist!</i></span>"
-	message_admins("[uppertext(ticker.mode.name)]: Selected [player] as a [role_text].")
-	if(istype(player.current, /mob/observer/dead))
-		create_default(player.current)
-	else
-		add_antagonist(player,0,0,0,1,1)
-	return 1
 
-/datum/antagonist/proc/build_candidate_list(var/ghosts_only)
-	// Get the raw list of potential players.
 	update_current_antag_max()
-	candidates = get_candidates(ghosts_only)
+	var/active_antags = get_active_antag_count()
+	log_debug("[uppertext(id)]: Found [active_antags]/[cur_max] active [role_text_plural].")
+
+	if(active_antags >= cur_max)
+		log_debug("Could not auto-spawn a [role_text], active antag limit reached.")
+		return 0
+
+	build_candidate_list(flags & (ANTAG_OVERRIDE_MOB|ANTAG_OVERRIDE_JOB))
+	if(!candidates.len)
+		log_debug("Could not auto-spawn a [role_text], no candidates found.")
+		return 0
+
+	attempt_spawn(1) //auto-spawn antags one at a time
```

Comment:
This *really* doesn't look like it should be there. Bad merge with something Polaris added?